Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -896,18 +896,24 @@ JSValue Interpreter::executeProgram(const SourceCode& source, JSGlobalObject*, J
                 }
             }
 
+            const Identifier& ident = JSONPPath.last().m_pathEntryName;
             if (JSONPPath.size() == 1 && JSONPPath.last().m_type != JSONPPathEntryTypeLookup) {
                 RELEASE_ASSERT(baseObject == globalObject);
                 JSGlobalLexicalEnvironment* scope = globalObject->globalLexicalEnvironment();
-                if (scope->hasProperty(globalObject, JSONPPath.last().m_pathEntryName))
+                if (scope->hasProperty(globalObject, ident)) {
```

Comment:
Could you please expand in commit message on why we want fast path to handle TDZ rather than tweaking JSONP parser not to take the fast path if there are `const` / `let`?